src/main/java/com/yelp/nrtsearch/server/handler/Handler.java

```diff
@@ -59,6 +62,41 @@ public StreamObserver<T> handle(StreamObserver<S> responseObserver) {
     throw new UnsupportedOperationException("This method is not supported");
   }
 
+  /**
+   * Handle a unary request, catching any exceptions and returning an error status to the client.
+   *
+   * @param name name of the request
+   * @param protoRequest request object
+   * @param responseObserver observer to send response to
+   * @param handler handler to process the request
+   * @param <T> request type
+   * @param <S> response type
+   */
+  public static <T extends GeneratedMessageV3, S extends GeneratedMessageV3>
+      void handleUnaryRequest(
+          String name, T protoRequest, StreamObserver<S> responseObserver, Handler<T, S> handler) {
+    try {
+      handler.handle(protoRequest, responseObserver);
```

Comment:
If doing this, do you want to add another handle method that returns the response and doesn't need responseObserver? It's currently passed just for error handling.